Source/diablo.cpp

```diff
@@ -1137,6 +1137,28 @@ void CreateLevel(lvl_entry lvldir)
 	}
 }
 
+void UnstuckChargers()

```

Comment:
I look at this method name and I already don't like it. Looks like a crazy hack to me.

Reply:
I think at least it's a short term solution.
It also fixes old vanilla saves or saves from previous deviltuionX versions.

Long term there could be other solutions like remodeling how the charging works (avoid missiles for it) or also save/load missiles when changing levels or ... but that's more complex and to stay savegame compatible with vanilla is not easy.

So I think this pr is currently fine. 🙂

Reply:
I hope it doesn't become one of those "multi-year short-term solutions" that I'm so used to seeing everywhere.

Reply:
That probably depends on how many/what other unsolved issues are left in a year.

Reply:
> I hope it doesn't become one of those "multi-year short-term solutions" that I'm so used to seeing everywhere.

I think that's possible. 😉 
But this PR doesn't span across the codebase or hinders future development or something like this.

Reply:
> But this PR doesn't span across the codebase or hinders future development or something like this.

It doesn't need to. If we keep introducing hacks to the codebase, even if they are all very localized and don't impact other things, there will reach a point where the code becomes a mess. These things pile up over time.

Anyways, I'll stop ranting.

Reply:
Better fixes are welcome, until that happens we have to be pragmatic

Reply:
> Better fixes are welcome, until that happens we have to be pragmatic

You make it sound like this is critical enough to warrant a hack, which is definitely not the case (IMHO). It is an extremely rare vanilla bug.

In my view, it would be more productive in the long term to pursue proper solutions instead of settling for hacks.

Reply:
I saw it happen just yesterday (after this pr was created) on souncaser83's stream, he didn't notice him self as he was looking elsewhere. I think it might be more common then we would think.

Reply:
> I think it might be more common then we would think.

I can see that being the case in speedruns where you are moving up/down stairs a lot with monsters around you. Maybe I didn't see it as much since I play more "casually", clearing floors and whatnot.

Even if that's the case, it is still largely inconsequential.

My overall point was that hacks should be weighed against how critical it is the thing that they are hacking around. If not critical enough, we should consider not applying the hack and living with the minor issue for a bit longer until a decent fix is made.

Reply:
I don't think we disagree much here, we just don't agree on what weights to insert in to the Bayesian model.